Label points (#496)
plotly/plotly.github.io: _posts/fundamentals/2019-05-26-how-to-label-points-with-text-in-a-scatter-plot.md

---
layout: tutorial-single_layout
title: Label Points with Text in a Scatter Plot
subtitle: A Step-by-Step Guide to Labeling Points with Text in a Scatter Plot
permalink: /label-points-with-text-in-a-scatter-plot/

> imageurl: /static/images/label-points-with-text/scatter-labeled-points.png

Reviewer: This image was removed in an earlier commit but it's necessary for the tutorial's thumbnail so we should add it back.

state: active
tags: textandlinks
meta_description: Add text labels to points in a scatter plot. Easily make interactive graphs online and for free with Chart Studio.
order: 5.5
actioncall: Make a scatter plot

> live-graph: <iframe width="900" height="800" frameborder="0" scrolling="no" src="https://plot.ly/~plotly2_demo/480.embed"></iframe>

Reviewer: I think we should update this chart so that it actually displays text values. At the moment it's kind of confusing.

steps:
 - title: Adding text labels to the points of scatter plot
   sub-steps:
    - copy: "After you've added the data to Chart Studio, head to 'Traces' under 'Style' menu in the left-hand side of Chart Studio. From 'Display' sub-panel click on the check box of 'Text' option.   " 

>       img: "![Label-points-with-text](../static/images/Text_label_.png)"

Reviewer: the link should is missing the label-text-with-points folder. it should be `../static/images/label-points-with-text/Text_label_.png`

>  - title: Associating text to each data point

Reviewer: A line with `sub-steps:` is necessary between the `title` and the `copy` line for this to build properly on Jekyll.

>     - copy: "To select a column associated to the text for each corresponding data points, head to 'Text' sub-panel and from the drop down menu of 'Text' property choose a column. You might define the position of the text on plot by heading to 'Text Position' property and choosing the preferred position from the drop down menu." 

Reviewer: I think can or may works better here.
```suggestion
    - copy: "To select a column associated to the text for each corresponding data points, head to 'Text' sub-panel and from the drop down menu of 'Text' property choose a column. You can define the position of the text on plot by heading to 'Text Position' property and choosing the preferred position from the drop down menu." 
```

>       img: "![Associate a text to each data point](../static/images/text_position.png)"

Reviewer: same link issue as above

>     - copy: "To position text values individually select the 'Custom' option, which will position them accordingly to the data position array you provide in the drop down menu."

Reviewer: This should be according
```suggestion
    - copy: "To position text values individually select the 'Custom' option, which will position them according to the data position array you provide in the drop down menu."
```

>       img: "![customize-text-labels](../static/images/custom_text_position.png)"

Reviewer: same link issue as above

meta_description: Add text labels to points in a scatter plot. Easily make interactive graphs online and for free with Chart Studio.
order: 5.5
actioncall: Make a scatter plot

live-graph: <iframe width="900" height="800" frameborder="0" scrolling="no" src="https://plot.ly/~PythonPlotBot/3391.embed"></iframe>

steps:
 - title: Adding text labels to the points of scatter plot
   sub-steps:
    - copy: "After you've added the data to Chart Studio, head to 'Traces' under 'Style' menu in the left-hand side of Chart Studio. From 'Display' sub-panel click on the check box of 'Text' option.   " 
      img: "![Label-points-with-text](../static/images/label-points-with-text/Text_label_.png)"
 - title: Associating text to each data point
   sub-steps:
    - copy: "To select a column associated to the text for each corresponding data points, head to 'Text' sub-panel and from the drop down menu of 'Text' property choose a column. You can define the position of the text on plot by heading to 'Text Position' property and choosing the preferred position from the drop down menu." 
      img: "![Associate a text to each data point](../static/images/label-points-with-text/Lable_position.png)"

>     - copy: "To position text values individually select the 'Custom' option, which will position them according to the data position array you provide in the drop down menu."

Reviewer: "data position array" is a little confusing to me. I think "column" is clearer.
```suggestion
    - copy: "To position text values individually select the 'Custom' option, which will position them according to the column selected in the drop down menu."
```
